airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -280,4 +301,41 @@ static String parseConnectorVersion(final String connectorImage) {
     return tokens[tokens.length - 1];
   }
 
+  /**
+   * Tests whether the provided JSON string represents a state message.
+   *
+   * @param input a JSON string that represents an {@link AirbyteMessage}.
+   * @return {@code true} if the message is a state message, {@code false} otherwise.
+   */
+  private static boolean isStateMessage(final String input) {
```

Comment:
i'm not sure i'm understanding this. if `Jsons.tryDeserialize` swallows an exception and returns an empty Optional for an invalid state message, then wouldn't the deserialized message not be present, and we'd return `false` for invalid state messages?

Reply:
@alovew The difference is that on this second try, we use a different class `AirbyteTypeMessage` which only deserializes the main `type` field, allowing us to see what type of message it is while avoiding deserializing the nested object (state, record, etc) that is causing the failure.